File: ordenacao-por-selecao/ordenacao-por-selecao-01.php
<?php 

function buscaMenor(array $arr) : int
{
    
    $menor = $arr[0];
    $menorIndice = 0;

    for($i = 1; $i < count($arr); $i++)
    {

        if($arr[$i] < $menor)
        {
            $menor = $arr[$i];
            $menorIndice = $i;
        }

    }

    return $menorIndice;
}

function ordenacaoPorSelecao(array $arr) : array
{
    $novoArr = [];
    $qtdArray = count($arr);

    for($i = 0; $i < $qtdArray; $i++){
        
        
        $menor = buscaMenor($arr);
        
        $novoArr[] = $arr[$menor];
        
        unset($arr[$menor]);

        $arr = array_values($arr); 

    }
    
    return $novoArr;
}

$array = [9,1,3,7];

$newArray = ordenacaoPorSelecao($array);

var_dump($newArray);

?>

File: ordenacao-por-selecao/ordenacao-por-selecao-01.py
def ordenacaoPorSelecao(arr):
    
    novoArr = []
    
    for j in range(len(arr)):
        menor = arr[0]
        menor_indice = 0
        for i in range(1, len(arr)):
            if arr[i] < menor:
                menor = arr[i]
                menor_indice = i
        novoArr.append(arr.pop(menor_indice))

    return novoArr         


arr = [5,6,7,3,2,1]


print (ordenacaoPorSelecao(arr))